keps/sig-apps/3328-add-queueName-to-jobSpec/README.md

```diff
@@ -0,0 +1,866 @@
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+This KEP aims to add a new field `queueName` to the spec of Job which helps unifying the
+queue API in Job.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+In Kubernetes world, Job plays an important role in batch scenarios, and generally it can be
+managed by a queue or something similar for resource management, multi-tenant support or
+special scheduling policies. But now we lack the native support of binding Jobs with a queue
+in core Kubernetes. We can see similar implementations in other batch systems outside the Kubernetes,
+e.g. in [Kubeflow](https://github.com/kubeflow/kubeflow),
+we can specify the queue name of the distributed training job in [`SchedulingPolicy`](https://github.com/kubeflow/common/blob/21f5ba8833a2e21df17601497a08396c9bae9ab2/pkg/apis/common/v1/types.go#L202-L209)
+We can also found the same implementation in [kubedl](https://github.com/kubedl-io/kubedl),
+another project helps to run deep learning workloads.
+
+[Kueue](https://github.com/kubernetes-sigs/kueue), a Kubernetes native Job queueing management is also
+calling out this feature. Currently a Kubernetes Job references to the kueue via the annotation,
+which is not native.
+
+So it's a good choice to support queueName in Job API, which helps to unify the usage for batch users.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+- Add a new field `queueName` to the spec of Job
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+- How to integrate with non-Kubernetes native Job controllers, e.g. TFJob Operator
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+Add a new field `queueName` to the spec of Job to unify the usage of batch users.
```

Comment:
Should we provide a server-side filter based on queueName, like we do for Pod.spec.nodeName? Although I'm not sure how useful it is. We don't need it in kueue.